### PR TITLE
Maven Package Version Configuration Fix

### LINKED ISSUE #747: [DOC] Add guidance on `FEATHR_MAVEN_VERSION` environment variable
### Willingness to contribute

Yes. I can contribute for this issue independently.

### URL(s) with the issue

_No response_

### Description of proposal (what needs changing)

This PR introduced a new environment variable `FEATHR_MAVEN_VERSION` https://github.com/feathr-ai/feathr/pull/746
We could add guidance of this config to other contributors after it's fully tested through release process.

### PR DESCRIPTION
**Pull request by Yuqing-cat**
Signed-off-by: Yuqing Wei <weiyuqing021@outlook.com>

## Description

Resolves #747  and fix FEATHR_MAVEN_VERSION not working. 

- Move logic to `version.py` as it is not a constant. 
- Update related lines in all 3 spark provider 
- Add configuration description in wiki.

## Does this PR introduce any user-facing changes?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure to clarify your proposed changes.

### CODE REVIEW

File: feathr_project/feathr/version.py [hunk 1]
```diff
@@ -1 +1,10 @@
-__version__ = "0.9.0-rc2"
\ No newline at end of file
+__version__ = "0.9.0-rc2"
+
+def get_version():
+    return __version__
+
+# Decouple Feathr MAVEN Version from Feathr Python SDK Version
+import os
+def get_maven_artifact():
```
Comment (xiaoyongzhu) [inline feathr_project/feathr/version.py:8]: Might be better if we call this `get_maven_artifact_version`?

Comment (Yuqing-cat) [reply feathr_project/feathr/version.py:8]: This method doesn't return the version name like `0.8.0` or `0.9.0-rc1` but the whole artifact name like `com.linkedin.feathr:feathr_2.12:0.8.0` for spark provider to use. I think `maven_artifact` would be more accurate.

File: feathr_project/feathr/version.py [hunk 2]
```diff
@@ -1 +1,10 @@
...
+    maven_version = os.environ.get("FEATHR_MAVEN_VERSION", __version__)
+    return f"com.linkedin.feathr:feathr_2.12:{maven_version}"
```
Comment (blrchen) [inline feathr_project/feathr/version.py:10]: Suggest following name change to make code more readable
- maven_version => maven_artifact_version 
- get_maven_artifact => get_maven_artifact_fullname
- FEATHR_MAVEN_VERSION =>  MAVEN_ARTIFACT_VERSION

File: docs/how-to-guides/feathr-configuration-and-env.md [hunk 3]
```diff
@@ -32,59 +32,65 @@ Feathr will get the configurations in the following order:
...
+| FEATURE_REGISTRY__PURVIEW__TYPE_SYSTEM_INITIALIZATION (Deprecated Soon) | Controls whether the type system (think this as the "schema" for the registry) will be initialized or not. Usually this is only required to be set to `True` to initialize schema, and then you can set it to `False` to shorten the initialization time.  | Required if using Purview directly without registry service. Deprecate soon, see [here](#deprecation) for more details. |
+| FEATHR_MAVEN_VERSION                                                    | Version number like `0.9.0`. Used to define maven package version when main jar is not defined.                                                                                                                                                            | Optional if custom maven package version is needed.                                                                     |
```
Comment (blrchen) [inline docs/how-to-guides/feathr-configuration-and-env.md:85]: Suggest change `Optional if custom maven package version is needed.` to just Optional

**Review by blrchen — APPROVED** [review]

**Review by xiaoyongzhu — APPROVED** [review]

**Pull request merged**